db/migrate/20200428143500_create_security_policy.rb

```diff
@@ -0,0 +1,55 @@
+# rake db:migrate:down VERSION=20200428000002
+# rake db:migrate:up VERSION=20200428000002
+class CreateSecurityPolicy < ActiveRecord::Migration[5.0]
+  def change
+    create_table :security_policies do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :description      
+      t.string     :ems_ref
+      t.references :ems,               :type => :bigint
+      t.references :cloud_tenant,      :type => :bigint   
+      t.integer    :sequence_number
+    end
+
+    add_index :security_policies,     :type
+
+    create_table :security_policy_rules do |t|
+      t.string     :type
+      t.string     :name
+      t.string     :description      
+      t.string     :ems_ref
+      t.references :ems,               :type => :bigint
+      t.references :cloud_tenant,      :type => :bigint   
+      t.references :security_policy,   :type => :bigint
+      t.integer    :sequence_number
+      t.string     :status
+      t.string     :action
+      t.string     :direction
+      t.string     :ip_protocol
+    end
```

Comment:
We have a "firewall_rules" table already I wonder how much overlap there is with that

Reply:
firewall_rules have a only 1 source and destination group. If you see a way to re-use the firewall_rules please advise.

Reply:
They look pretty similar:
```
#<FirewallRule:0x000055e26552bbe0
 id: 65000000000001,
 name: nil,
 display_name: nil,
 group: nil,
 enabled: nil,
 required: nil,
 host_protocol: "TCP",
 direction: "inbound",
 port: 0,
 end_port: 65535,
 created_on: Wed, 29 Apr 2020 17:30:31 UTC +00:00,
 updated_on: Wed, 29 Apr 2020 17:30:31 UTC +00:00,
 resource_id: 65000000001009,
 resource_type: "SecurityGroup",
 source_security_group_id: 65000000001009,
 source_ip_range: nil,
 ems_ref: nil,
 network_protocol: nil>
```

```
#<ManageIQ::Providers::Nsxt::NetworkManager::SecurityPolicyRule:0x0000563f18600d30
 id: 65000000000783,
 type: "ManageIQ::Providers::Nsxt::NetworkManager::SecurityPolicyRule",
 name: "DHCP-all",
 description: nil,
 ems_ref: "1717e720-6a28-11ea-98a6-5d1d088ebb0c",
 ems_id: 65000000000057,
 cloud_tenant_id: nil,
 security_policy_id: 65000000000512,
 sequence_number: 10,
 status: "active",
 action: "ALLOW",
 direction: "IN_OUT",
 ip_protocol: "IPV4_IPV6">
```

Could we add anything that the existing firewall_rules table is missing?